DataFormats/FTLRecHit/src/FTLRecHit.cc

```diff
@@ -0,0 +1,111 @@
+#include "DataFormats/FTLRecHit/interface/FTLRecHit.h"
+#include "DataFormats/ForwardDetId/interface/FastTimeDetId.h"
+#include <cassert>
+#include <math.h>
+
+FTLRecHit::FTLRecHit() : CaloRecHit(), flagBits_(0) {
+}
+
+FTLRecHit::FTLRecHit(const DetId& id, float energy, float time, uint32_t flags, uint32_t flagBits) :
+  CaloRecHit(id,energy,time,flags), flagBits_(flagBits) {
+}
+
+float FTLRecHit::chi2() const {
+  uint32_t rawChi2 = 0x7F & (flags()>>4);
+  return (float)rawChi2 / (float)((1<<7)-1) * 64.;
```

Comment:
This function is never called due to the present simplistic state of the reconstruction (this is more or less a copy of digi values).
It will be rewritten anyway before it is actually used.